### PR TITLE
jsrt: JsDiagEvaluate uses JsValueRef for expression

### LINKED ISSUE #2282: xplat: JsDiagEvaluate signature
Following up #2251, we plan to change `JsDiagEvaluate` signature to take `JsValueRef` string, and remove `JsDiagEvaluateUtf8`.

//cc: @liminzhu @agarwal-sandeep @obastemur

### PR DESCRIPTION
**Pull request by obastemur**
Fixes #2282

### CODE REVIEW

File: lib/Jsrt/JsrtDiag.cpp [hunk 1]
```diff
@@ -670,15 +670,57 @@ CHAKRA_API JsDiagGetObjectFromHandle(
...
+            isString = Js::JavascriptString::Is(expressionVal);
+            if (!isString)
+            {
+                return JsErrorInvalidArgument;
+            }
+        }
+
+        const size_t len = isExternalArray ?
+            ((Js::ExternalArrayBuffer*)(expressionVal))->GetByteLength() :
+            ((Js::JavascriptString*)(expressionVal))->GetLength();
```
Comment (agarwal-sandeep) [inline lib/Jsrt/JsrtDiag.cpp]: Please use Js::ExternalArrayBuffer::FromVar and Js::JavascriptString::FromVar

File: bin/ch/Debugger.cpp [hunk 2]
```diff
@@ -174,10 +174,7 @@ JsValueRef Debugger::Evaluate(JsValueRef callee, bool isConstructCall, JsValueRe
     if (argumentCount > 2)
     {
         IfJsErrorFailLogAndRet(ChakraRTInterface::JsNumberToInt(arguments[1], &stackFrameIndex));
-
-        AutoString argstr(arguments[2]);
-        IfJsErrorFailLogAndRet(argstr.GetError());
-        ChakraRTInterface::JsDiagEvaluateUtf8(argstr.GetString(), stackFrameIndex, &result);
+        ChakraRTInterface::JsDiagEvaluate(arguments[2], stackFrameIndex, JsParseScriptAttributeNone, &result);
```
Comment (agarwal-sandeep) [inline bin/ch/Debugger.cpp]: >JsDiagEvaluate [](start = 27, length = 14)

We need to add test for covering JsDiagEvaluate where 1st argument is ExternalArrayBuffer.
The way we have ch test debugger written Debugger::Evaluate is called from js file so argument is already JsValueRef (JavascriptString). One way to test external buffer is to add a new argument to evaluate which will convert string to external buffer.

**Comment by agarwal-sandeep** [comment]
@digitalinfinity @liminzhu FYI that this will change the JsDiagEvaluate API signature and we will only have 1 API which takes JsValueRef. Windows version which takes wchar is removed.

File: lib/Jsrt/JsrtCommonExports.inc [hunk 3]
```diff
@@ -115,5 +115,4 @@
     JsRunSerialized
     JsCreatePropertyId
     JsCopyPropertyId
```
Comment (jianchun) [inline lib/Jsrt/JsrtCommonExports.inc]: Unrelated... was searching where "JsDiagEvaluate" was exported, found in "bin/ChakraCore/ChakraCore.def". Looks these are supposed to be in that file as well?

Comment (obastemur) [inline lib/Jsrt/JsrtCommonExports.inc]: Static Lib?

File: bin/ch/Debugger.cpp (hunk 2 above)
Comment (obastemur) [inline bin/ch/Debugger.cpp]: @agarwal-sandeep there is one additional test that force uses ArrayBuffer.

**Comment by obastemur** [comment]
ping. Good to go ?

**Comment by agarwal-sandeep** [comment]
LGTM

File: lib/Jsrt/JsrtCommonExports.inc (hunk 3 above)
Comment (jianchun) [inline lib/Jsrt/JsrtCommonExports.inc]: @obastemur If that's the case, that indicates "JsDiagEvaluate" and all other APIs there are unavailable in static lib?

**Comment by liminzhu** [comment]
LGTM

File: lib/Jsrt/JsrtCommonExports.inc (hunk 3 above)
Comment (obastemur) [inline lib/Jsrt/JsrtCommonExports.inc]: Windows static lib support is couple of changes away. Once its there, JsDiagEvaluate might be better defined with the others. Good point anyways, Thanks!

**Comment by obastemur** [comment]
@agarwal-sandeep @jianchun @liminzhu Thanks for the reviews

**Pull request merged**